dataStructuresObjectsAndArrays/Reversing an array.js

```diff
@@ -0,0 +1,62 @@
+/*
+Arrays have a reverse method that changes the array by inverting the order in which
+ its elements appear. For this exercise, write two functions, reverseArray and 
+reverseArrayInPlace. The first, reverseArray, takes an array as argument and produces
+ a new array that has the same elements in the inverse order. The second, 
+reverseArrayInPlace, does what the reverse method does: it modifies the array given 
+as argument by reversing its elements. Neither may use the standard reverse method.
+
+Thinking back to the notes about side effects and pure functions in the previous 
+chapter, which variant do you expect to be useful in more situations? Which one runs faster?
+*/
+
+function reverseArray(arr) {
+  const newArr = [];
+  for (let i = arr.length - 1; i >= 0; i--) {
+    newArr.push(arr[i]);
+  }
+  return newArr;
+}
+
+function reverseArrayInPlace(array) {
+  for (let h = 0; h < Math.floor(array.length / 2); h++) {
+    let a = array[h];
+    array[h] = array[array.length - 1 - h];
+    array[array.length - 1 - h] = a;
+  }
```

Comment:
Function missing return statement.
**Tips:** you can create `arrayLenght` variable to store `array.length` in it before `for` loop